wormhole-connect/src/views/v2/Redeem/index.tsx

```diff
@@ -575,59 +650,51 @@ const Redeem = () => {
         TransferWallet.RECEIVING,
       );
 
-      let receipt: routes.Receipt | undefined;
-
-      if (isTxDestQueued && routes.isFinalizable(route)) {
-        receipt = await route.finalize(signer, routeContext.receipt);
-      } else if (!isTxDestQueued && routes.isManual(route)) {
-        receipt = await route.complete(signer, routeContext.receipt);
-      }
-
-      if (!receipt || (!isRedeemed(receipt) && !isCompleted(receipt))) {
-        throw new Error('Transfer not completed');
-      }
-
-      if (receipt.destinationTxs && receipt.destinationTxs.length > 0) {
-        txId = receipt.destinationTxs[receipt.destinationTxs.length - 1].txid;
+      const finishPromise = (() => {
+        if (isTxDestQueued && routes.isFinalizable(route)) {
+          return route.finalize(signer, routeContext.receipt);
+        } else if (!isTxDestQueued && routes.isManual(route)) {
+          return route.complete(signer, routeContext.receipt);
+        } else {
+          // Should be unreachable
+          return undefined;
+        }
+      })();
+
+      if (finishPromise) {
+        config.triggerEvent({
+          type: 'transfer.redeem.start',
+          details,
+        });
+
+        // Await this promise just so that we catch any errors thrown by it and handle them below
+        await finishPromise;
       }
-
-      config.triggerEvent({
-        type: 'transfer.redeem.start',
-        details: transferDetails,
-      });
-
-      setIsClaimInProgress(false);
-      setClaimError('');
     } catch (e: any) {
-      const [uiError, transferError] = interpretTransferError(e, toChain);
-
-      setClaimError(uiError);
-
-      config.triggerEvent({
-        type: 'transfer.redeem.error',
-        details: transferDetails,
-        error: transferError,
-      });
-
+      // This could be all kinds of unexpected errors
+      // Kick it up to the main useEffect where we handle receipt state changes
+      setUnhandledManualClaimError(e);
       setIsClaimInProgress(false);
-      console.error(e);
-    }
-    if (txId !== undefined) {
-      config.triggerEvent({
-        type: 'transfer.redeem.success',
-        details: transferDetails,
-      });
     }
   };
 
   // Main CTA button which has separate states for automatic and manual claims
   const actionButton = useMemo(() => {
-    if (
-      !isTxComplete &&
-      !isTxRefunded &&
-      !isTxFailed &&
-      (isTxDestQueued || !isAutomaticRoute)
-    ) {
+    if (isTxComplete || isTxRefunded) {
+      return (
+        <Button
+          variant="primary"
+          className={classes.actionButton}
+          onClick={() => {
+            dispatch(setRoute('bridge'));
+          }}
+        >
+          <Typography textTransform="none">Start a new transaction</Typography>
+        </Button>
+      );
+    }
+
+    if (isTxDestQueued || !isAutomaticRoute) {
       if (isTxAttested && !isConnectedToReceivingWallet) {
```

Comment:
When `isTxDestQueued === true`, then `isTxAttested === false` and users won't be able to connect their receiving wallet. I think we should change `isTxAttested` back to `>= TransferState.Attested` or figure out something better

Reply:
I believe we only need to connect wallet in Redeem view when the transaction is in ready to claim state.

Reply:
The tx may be in a ready to claim state while isTxDestQueued is true

Reply:
I dont like using gte on the receipt state because it's unclear and "happens to work". I'd like to be more explicit. how does this look @kev1n-peters ? https://github.com/wormhole-foundation/wormhole-connect/pull/2860/commits/48b72828a958620a6e8b066a447c671f190116c4

Reply:
That looks easier to read to me